src/components/CategoryPicker.tsx

```diff
@@ -42,44 +56,40 @@ function CategoryPicker({selectedCategory, policyCategories, policyRecentlyUsedC
             false,
             false,
             true,
-            policyCategories,
+            policyCategories ?? {},
             validPolicyRecentlyUsedCategories,
             false,
         );
 
-        const header = OptionsListUtils.getHeaderMessageForNonUserList(lodashGet(categoryOptions, '[0].data', []).length > 0, debouncedSearchValue);
-        const policiesCount = OptionsListUtils.getEnabledCategoriesCount(_.values(policyCategories));
-        const isCategoriesCountBelowThreshold = policyCategoriesCount < CONST.CATEGORY_LIST_THRESHOLD;
+        const categoryData = categoryOptions?.[0]?.data ?? [];
+        const header = OptionsListUtils.getHeaderMessageForNonUserList(categoryData.length > 0, debouncedSearchValue);
+        const policiesCount = OptionsListUtils.getEnabledCategoriesCount(policyCategories ?? {});
+        const isCategoriesCountBelowThreshold = policiesCount < CONST.CATEGORY_LIST_THRESHOLD;
         const showInput = !isCategoriesCountBelowThreshold;
 
-        return [categoryOptions, header, policiesCount, showInput];
-    }, [policyCategories, policyRecentlyUsedCategories, debouncedSearchValue, selectedOptions]);
+        const selectedKey = categoryData.filter((category) => category.searchText === selectedCategory)[0]?.keyForList;
```

Comment:
Any reason in particular for removing the memoization we had for this?

Reply:
Hmm this memoization is still there. We just have it in one `useMemo` hook instead of 2 separate `useMemo`s. The second one previously depended on the first one and both of them had `selectedCategory` in deps.

Reply:
Ahh actually I moved it automatically but when I look at the deps of both hooks it looks like it's better to keep it the old way. I will change it to the previous version.

Reply:
Done 👌